src/gsy_e/models/market/forward.py

```diff
@@ -0,0 +1,138 @@
+                 readonly: bool = False,
+                 grid_fee_type: int = ConstSettings.MASettings.GRID_FEE_TYPE,
+                 grid_fees: Optional[GridFee] = None,
+                 name: Optional[str] = None) -> None:
+        # pylint: disable=too-many-arguments
+        if not ConstSettings.ForwardMarketSettings.ENABLE_FORWARD_MARKETS:
+            return
+
+        super().__init__(bc=bc, notification_listener=notification_listener,
+                         readonly=readonly, grid_fee_type=grid_fee_type,
+                         grid_fees=grid_fees, name=name)
+
+    @staticmethod
+    @abstractmethod
+    def _get_start_time(current_time: DateTime) -> DateTime:
+        """Return time when the market block starts."""
+
+    @staticmethod
+    @abstractmethod
+    def _get_end_time(current_time: DateTime) -> DateTime:
+        """Return time when the market block ends."""
+
+    @staticmethod
+    @abstractmethod
+    def _get_market_slot_duration(current_time: DateTime, _config) -> duration:
+        """Return duration of market slots inside the market block."""
+
+    def create_future_market_slots(self, current_market_time_slot: DateTime,
+                                   config: "SimulationConfig") -> None:
+        if not ConstSettings.ForwardMarketSettings.ENABLE_FORWARD_MARKETS:
+            return
+        self._create_future_market_slots(self._get_start_time(current_market_time_slot),
+                                         self._get_end_time(current_market_time_slot), config)
+
+
+class DayForwardMarket(ForwardMarketBase):
+    """Day forward market implementation"""
+
+    @staticmethod
+    def _get_start_time(current_time: DateTime) -> DateTime:
+        return current_time.set(hour=0, minute=0).add(days=1)
+
+    @staticmethod
+    def _get_end_time(current_time: DateTime) -> DateTime:
+        return current_time.set(hour=0, minute=0).add(weeks=1, days=1, hours=-1)
+
+    @staticmethod
+    def _get_market_slot_duration(_current_time: DateTime, _config) -> duration:
+        return duration(hours=1)
+
+
+class WeekForwardMarket(ForwardMarketBase):
+    """Week forward market implementation"""
+
+    @staticmethod
+    def _get_start_time(current_time: DateTime) -> DateTime:
+        days_until_next_monday = 7 - (current_time.day_of_week - 1)
+        return current_time.set(hour=0, minute=0).add(days=days_until_next_monday)
+
+    @staticmethod
+    def _get_end_time(current_time: DateTime) -> DateTime:
+        return current_time.set(hour=0, minute=0).add(years=1)
+
+    @staticmethod
+    def _get_market_slot_duration(_current_time: DateTime, _config) -> duration:
+        return duration(weeks=1)
+
+
+class MonthForwardMarket(ForwardMarketBase):
+    """Month forward market implementation"""
+
+    @staticmethod
+    def _get_start_time(current_time: DateTime) -> DateTime:
+        return current_time.set(day=1, hour=0, minute=0).add(months=1)
```

Comment:
Change the order of `.set` and `.add` as discussed below in `YearForwardMarket`. Thanks